rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -188,50 +187,106 @@ static int create_enclave_range(int dev_fd, uint64_t mmap_size,
 	return 0;
 }
 
-static bool encl_create(int dev_fd, unsigned long bin_size,
-			struct sgx_secs *secs, struct enclave_info *encl_info,
-			struct metadata *meta_data,
-			struct sgx_sigstruct *sigstruct)
+/* Sanity check attributes to prevent possible tampering */
+static bool check_sgx_attributes(const uint64_t req_attrs,
+				 const uint64_t req_attrs_mask)
 {
-	struct sgx_enclave_create ioc;
-	uint64_t xfrm;
+	/* skeleton doesn't support 32-bit mode */
+	uint64_t enforced_pattern = SGX_ATTR_MODE64BIT;
 
-	memset(secs, 0, sizeof(*secs));
-	secs->attributes = SGX_ATTR_MODE64BIT;
-	if (enclave_debug)
-		secs->attributes |= SGX_ATTR_DEBUG;
-	/* Check attributes to prevent possible tampering of metadata area */
-	if ((meta_data->attributes & sigstruct->body.attributes) !=
-	    sigstruct->body.attributes) {
-		fprintf(stderr, "Invalid attributes value.\n");
+	if (req_attrs & ~SGX_ATTR_ALLOWED_MASK) {
+		fprintf(stderr,
+			"Invalid attributes value. The unsupported attributes %#lx are set.\n",
+			req_attrs & ~SGX_ATTR_ALLOWED_MASK);
+		return false;
+	}
+
+	if (req_attrs_mask & ~SGX_ATTR_ALLOWED_MASK) {
```

Comment:
I am a little confused about this check.
SGX_ATTR_ALLOWED_MASK = 1011 0110
Assum user define:  
- req_attrs = 0000 0110, which means Debug enclave
- req_attrs_mask = 1111 1111, which means author want secs->attrs equals to req_attrs 0000 0110
I think this reabsonble.
But this check will not pass and print error.
Please tell me if my thought is wrong.

Reply:
You think the mask doesn't have any semantics on the definition of SECS attributes. It is just a bitmap.

You are right. Excellent!